amf-shapes/shared/src/main/scala/amf/shapes/internal/spec/jsonschema/semanticjsonschema/transform/PropertyShapeTransformer.scala

```diff
@@ -34,11 +32,12 @@ case class PropertyShapeTransformer(property: PropertyShape, ctx: ShapeTransform
     propertyMapping
   }
 
-  private def transformArray(array: ArrayShape) = {
+  private def transformArray(array: ArrayShape): Unit = {
     propertyMapping.withAllowMultiple(true)
     array.items match {
```

Comment:
Can't we just call the root schema transformation here? A possible blocker is that we don't accept ScalarShapes as roots but can this be a possible case?

Reply:
Yes, that is the problem to call the root. Also calling the root transformation will extract the mapping to a declaration, for scalars this makes no sense.